### ISSUE TITLE
Allow widgets to be placed across multiple browser windows

### ISSUE THREAD

**Issue by veggiesaurus** [CARTAvis/carta-frontend#2003]
As requested by a number of users, it would be great to allow CARTA to operate over multiple browser windows for situations where a user is running a multi-screen config. 

I think this is a useful request, and is commonly supported by a number of technical applications. However, it's particularly difficult to achieve in CARTA, because of its browser-based architecture. Each window is sandboxed from the other, so we would need a complex message-passing approach for this. However, there are ways to do this...e.g. [golden-layout](https://golden-layout.com/tutorials/working-with-popouts.html) example.

**Comment by veggiesaurus** [comment]
I don't think this is going to be tackled anytime soon, but I thought  I'd just make this issue for tracking the discussion itself. The technical overhead for such a feature is unfortunately pretty large.

**Comment by kswang1029** [comment]
It seems possible to utilize golden-layout's function to detach a widget as a new browser window. 

Try 
https://golden-layout.com
and click the left-most button in the top-right corner of a widget. 

With this feature, I think users can have choices to display their widgets elsewhere in their viewable screens, while keep other sets of widgets organized in the main browser window.

**Comment by veggiesaurus** [comment]
> It seems possible to utilize golden-layout's function to detach a widget as a new browser window.
> 
> Try https://golden-layout.com and click the left-most button in the top-right corner of a widget.
> 
> With this feature, I think users can have choices to display their widgets elsewhere in their viewable screens, while keep other sets of widgets organized in the main browser window.

We need to be pretty careful with this for a couple of reasons:
1. Golden layout functionality is pretty limited in scope, and hasn't been updated for a long time.
2. We need to be careful in terms of _what_ we allow in the second window, because we will need to use message posting to transfer state.
